src/afl-fuzz-run.c

```diff
@@ -65,21 +65,40 @@ void write_to_testcase(afl_state_t *afl, void *mem, u32 len) {
 
 #endif
 
-  if (unlikely(afl->mutator && afl->mutator->afl_custom_pre_save)) {
+  bool has_custom_pre_save = false;
+
+  if (unlikely(afl->number_of_custom_mutators)) {
 
     u8 *new_buf = NULL;
+    ssize_t new_size;
+
+    LIST_FOREACH(&afl->custom_mutator_list, struct custom_mutator, {
+
+      if (el->afl_custom_pre_save) {
+        new_size = el->afl_custom_pre_save(
+          el->data, mem, len, &new_buf
```

Comment:
next round should get `new_buf` and `new_size` of last round.
Or do we only want to allow a single valid `pre_save` hook?